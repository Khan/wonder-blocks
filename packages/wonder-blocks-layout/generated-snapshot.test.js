// @flow
// This file is auto-generated by gen-snapshot-tests.js
// Do not edit this file.  To make changes to these snapshot tests:
//   1. edit the markdown documentation files in the package,
//        packages/wonder-blocks-layout
//   2. Run `yarn run gen-snapshot-tests`.
import React from "react";
import renderer from "react-test-renderer";

// Mock react-dom as jest doesn't like findDOMNode.
jest.mock("react-dom");

describe("wonder-blocks-layout", () => {
    it("example 1", () => {
        const {StyleSheet} = require("aphrodite");
        const {View, addStyle} = require("@khanacademy/wonder-blocks-core");
        const Color = require("@khanacademy/wonder-blocks-color").default;
        const Spacing = require("@khanacademy/wonder-blocks-spacing").default;
        const {Spring, Strut} = require("./index.js");

        // TODO(kevinb): replace with wonder-blocks-button once yarn workspaces land
        const Button = addStyle("button");

        const styles = StyleSheet.create({
            container: {
                flexDirection: "row",
                border: `solid 1px ${Color.offBlack50}`,
            },
            button: {
                fontSize: 24,
                width: 100,
                borderRadius: 4,
                border: "none",
                height: 40,
            },
        });

        const example = (
            <View style={styles.container}>
                <Button style={styles.button}>A</Button>
                <Strut size={Spacing.small} />
                <Button style={styles.button}>B</Button>
                <Strut size={Spacing.small} />
                <Button style={styles.button}>C</Button>
                <Spring />
                <Button style={styles.button}>Cancel</Button>
                <Strut size={Spacing.small} />
                <Button style={styles.button}>Accept</Button>
            </View>
        );
        const tree = renderer.create(example).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
