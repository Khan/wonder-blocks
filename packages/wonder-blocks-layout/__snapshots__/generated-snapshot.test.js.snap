// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`wonder-blocks-layout example 1 1`] = `
<div
  className=""
  style={
    Object {
      "alignItems": "stretch",
      "border": "solid 1px rgba(33,36,44,0.50)",
      "borderStyle": "solid",
      "borderWidth": 0,
      "boxSizing": "border-box",
      "display": "flex",
      "flexDirection": "row",
      "margin": 0,
      "minHeight": 0,
      "minWidth": 0,
      "padding": 0,
      "position": "relative",
      "zIndex": 0,
    }
  }
>
  <button
    className=""
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchCancel={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "::MozFocusInner": Object {
          "border": 0,
        },
        "alignItems": "center",
        "background": "#1865f2",
        "border": "none",
        "borderRadius": 4,
        "boxSizing": "border-box",
        "color": "#ffffff",
        "cursor": "pointer",
        "display": "inline-flex",
        "height": 40,
        "justifyContent": "center",
        "margin": 0,
        "outline": "none",
        "paddingBottom": 0,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 0,
        "position": "relative",
        "textDecoration": "none",
        "width": 100,
      }
    }
    tabIndex={0}
  >
    <span
      className=""
      style={
        Object {
          "MozOsxFontSmoothing": "grayscale",
          "WebkitFontSmoothing": "antialiased",
          "display": "block",
          "fontFamily": "Lato, sans-serif",
          "fontSize": 16,
          "fontWeight": "bold",
          "lineHeight": "20px",
          "pointerEvents": "none",
        }
      }
    >
      A
    </span>
  </button>
  <div
    aria-hidden="true"
    className=""
    style={
      Object {
        "MsFlexBasis": 12,
        "MsFlexPreferredSize": 12,
        "WebkitFlexBasis": 12,
        "alignItems": "stretch",
        "borderStyle": "solid",
        "borderWidth": 0,
        "boxSizing": "border-box",
        "display": "flex",
        "flexBasis": 12,
        "flexDirection": "column",
        "flexShrink": 0,
        "margin": 0,
        "minHeight": 0,
        "minWidth": 0,
        "padding": 0,
        "position": "relative",
        "zIndex": 0,
      }
    }
  />
  <button
    className=""
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchCancel={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "::MozFocusInner": Object {
          "border": 0,
        },
        "alignItems": "center",
        "background": "#1865f2",
        "border": "none",
        "borderRadius": 4,
        "boxSizing": "border-box",
        "color": "#ffffff",
        "cursor": "pointer",
        "display": "inline-flex",
        "height": 40,
        "justifyContent": "center",
        "margin": 0,
        "outline": "none",
        "paddingBottom": 0,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 0,
        "position": "relative",
        "textDecoration": "none",
        "width": 100,
      }
    }
    tabIndex={0}
  >
    <span
      className=""
      style={
        Object {
          "MozOsxFontSmoothing": "grayscale",
          "WebkitFontSmoothing": "antialiased",
          "display": "block",
          "fontFamily": "Lato, sans-serif",
          "fontSize": 16,
          "fontWeight": "bold",
          "lineHeight": "20px",
          "pointerEvents": "none",
        }
      }
    >
      B
    </span>
  </button>
  <div
    aria-hidden="true"
    className=""
    style={
      Object {
        "MsFlexBasis": 12,
        "MsFlexPreferredSize": 12,
        "WebkitFlexBasis": 12,
        "alignItems": "stretch",
        "borderStyle": "solid",
        "borderWidth": 0,
        "boxSizing": "border-box",
        "display": "flex",
        "flexBasis": 12,
        "flexDirection": "column",
        "flexShrink": 0,
        "margin": 0,
        "minHeight": 0,
        "minWidth": 0,
        "padding": 0,
        "position": "relative",
        "zIndex": 0,
      }
    }
  />
  <button
    className=""
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchCancel={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "::MozFocusInner": Object {
          "border": 0,
        },
        "alignItems": "center",
        "background": "#1865f2",
        "border": "none",
        "borderRadius": 4,
        "boxSizing": "border-box",
        "color": "#ffffff",
        "cursor": "pointer",
        "display": "inline-flex",
        "height": 40,
        "justifyContent": "center",
        "margin": 0,
        "outline": "none",
        "paddingBottom": 0,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 0,
        "position": "relative",
        "textDecoration": "none",
        "width": 100,
      }
    }
    tabIndex={0}
  >
    <span
      className=""
      style={
        Object {
          "MozOsxFontSmoothing": "grayscale",
          "WebkitFontSmoothing": "antialiased",
          "display": "block",
          "fontFamily": "Lato, sans-serif",
          "fontSize": 16,
          "fontWeight": "bold",
          "lineHeight": "20px",
          "pointerEvents": "none",
        }
      }
    >
      C
    </span>
  </button>
  <div
    aria-hidden="true"
    className=""
    style={
      Object {
        "alignItems": "stretch",
        "borderStyle": "solid",
        "borderWidth": 0,
        "boxSizing": "border-box",
        "display": "flex",
        "flexDirection": "column",
        "flexGrow": 1,
        "margin": 0,
        "minHeight": 0,
        "minWidth": 0,
        "padding": 0,
        "position": "relative",
        "zIndex": 0,
      }
    }
  />
  <button
    className=""
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchCancel={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "::MozFocusInner": Object {
          "border": 0,
        },
        "alignItems": "center",
        "background": "#1865f2",
        "border": "none",
        "borderRadius": 4,
        "boxSizing": "border-box",
        "color": "#ffffff",
        "cursor": "pointer",
        "display": "inline-flex",
        "height": 40,
        "justifyContent": "center",
        "margin": 0,
        "outline": "none",
        "paddingBottom": 0,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 0,
        "position": "relative",
        "textDecoration": "none",
        "width": 100,
      }
    }
    tabIndex={0}
  >
    <span
      className=""
      style={
        Object {
          "MozOsxFontSmoothing": "grayscale",
          "WebkitFontSmoothing": "antialiased",
          "display": "block",
          "fontFamily": "Lato, sans-serif",
          "fontSize": 16,
          "fontWeight": "bold",
          "lineHeight": "20px",
          "pointerEvents": "none",
        }
      }
    >
      Cancel
    </span>
  </button>
  <div
    aria-hidden="true"
    className=""
    style={
      Object {
        "MsFlexBasis": 12,
        "MsFlexPreferredSize": 12,
        "WebkitFlexBasis": 12,
        "alignItems": "stretch",
        "borderStyle": "solid",
        "borderWidth": 0,
        "boxSizing": "border-box",
        "display": "flex",
        "flexBasis": 12,
        "flexDirection": "column",
        "flexShrink": 0,
        "margin": 0,
        "minHeight": 0,
        "minWidth": 0,
        "padding": 0,
        "position": "relative",
        "zIndex": 0,
      }
    }
  />
  <button
    className=""
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchCancel={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "::MozFocusInner": Object {
          "border": 0,
        },
        "alignItems": "center",
        "background": "#1865f2",
        "border": "none",
        "borderRadius": 4,
        "boxSizing": "border-box",
        "color": "#ffffff",
        "cursor": "pointer",
        "display": "inline-flex",
        "height": 40,
        "justifyContent": "center",
        "margin": 0,
        "outline": "none",
        "paddingBottom": 0,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 0,
        "position": "relative",
        "textDecoration": "none",
        "width": 100,
      }
    }
    tabIndex={0}
  >
    <span
      className=""
      style={
        Object {
          "MozOsxFontSmoothing": "grayscale",
          "WebkitFontSmoothing": "antialiased",
          "display": "block",
          "fontFamily": "Lato, sans-serif",
          "fontSize": 16,
          "fontWeight": "bold",
          "lineHeight": "20px",
          "pointerEvents": "none",
        }
      }
    >
      Accept
    </span>
  </button>
</div>
`;
